server/src/main/java/org/elasticsearch/rest/action/search/SearchResponseMetrics.java

```diff
@@ -8,31 +8,55 @@
 
 package org.elasticsearch.rest.action.search;
 
+import org.elasticsearch.telemetry.metric.LongCounter;
 import org.elasticsearch.telemetry.metric.LongHistogram;
 import org.elasticsearch.telemetry.metric.MeterRegistry;
 
+import java.util.Map;
+
 public class SearchResponseMetrics {
 
+    public enum ResponseCountTotalStatus {
+        SUCCESS,
+        PARTIAL_FAILURE,
+        FAILURE
+    }
+
+    public static final String RESPONSE_COUNT_TOTAL_STATUS_ATTRIBUTE_NAME = "status";
+
     public static final String TOOK_DURATION_TOTAL_HISTOGRAM_NAME = "es.search_response.took_durations.histogram";
+    public static final String RESPONSE_COUNT_TOTAL_COUNTER_NAME = "es.search_response.response_count.total";
 
     private final LongHistogram tookDurationTotalMillisHistogram;
+    private final LongCounter responseCountTotalCounter;
 
     public SearchResponseMetrics(MeterRegistry meterRegistry) {
         this(
             meterRegistry.registerLongHistogram(
                 TOOK_DURATION_TOTAL_HISTOGRAM_NAME,
                 "The SearchResponse.took durations in milliseconds, expressed as a histogram",
                 "millis"
+            ),
+            meterRegistry.registerLongCounter(
+                RESPONSE_COUNT_TOTAL_COUNTER_NAME,
+                "The cumulative total of search responses with an attribute to describe"
+                    + "success, partial failure, or failure, expressed as a counter and an attribute",
+                "count"
             )
         );
     }
 
-    private SearchResponseMetrics(LongHistogram tookDurationTotalMillisHistogram) {
+    private SearchResponseMetrics(LongHistogram tookDurationTotalMillisHistogram, LongCounter responseCountTotalCounter) {
         this.tookDurationTotalMillisHistogram = tookDurationTotalMillisHistogram;
+        this.responseCountTotalCounter = responseCountTotalCounter;
     }
 
     public long recordTookTime(long tookTime) {
         tookDurationTotalMillisHistogram.record(tookTime);
         return tookTime;
     }
+
+    public void incrementResponseCount(ResponseCountTotalStatus responseCountTotalStatus) {
+        responseCountTotalCounter.incrementBy(1L, Map.of(RESPONSE_COUNT_TOTAL_STATUS_ATTRIBUTE_NAME, responseCountTotalStatus));
+    }
```

Comment:
This is what's causing the following build failure:

https://buildkite.com/elastic/elasticsearch-pull-request/builds/23506#01903d0a-f3e5-4f38-8025-68037928d91b
 
 ```
 Caused by: java.lang.IllegalArgumentException: attributes do not support value type of [org.elasticsearch.rest.action.search.SearchResponseMetrics.ResponseCountTotalStatus]
»       at org.elasticsearch.telemetry.apm@8.15.0-SNAPSHOT/org.elasticsearch.telemetry.apm.internal.metrics.OtelHelper.lambda$fromMap$0(OtelHelper.java:50)
```


You could change this to:

```
responseCountTotalCounter.incrementBy(1L, Map.of(RESPONSE_COUNT_TOTAL_STATUS_ATTRIBUTE_NAME, responseCountTotalStatus.name()));
}
```

Or,

```

    public enum ResponseCountTotalStatus {
        SUCCESS("success"),
        PARTIAL_FAILURE("partial_failure"),
        FAILURE("failure");
        private final String value;

        ResponseCountTotalStatus(String value) {
            this.value = value;
        }

        public String value() {
            return value
        }
    }
    ...
    
        responseCountTotalCounter.incrementBy(1L, Map.of(RESPONSE_COUNT_TOTAL_STATUS_ATTRIBUTE_NAME, responseCountTotalStatus.value()));

   ```

Reply:
After chatting with @original-brownbear my concerns around the assert triggered by this error have been alleviated. My understanding is the assert is related to ensuring we don't except any exceptions to be triggered as part of `onResponse` except for unusual cases. I will fix this and then expect the tests will pass.

Reply:
Changed to your suggestion of `value` but used `displayName` instead to not overload the term `value`.